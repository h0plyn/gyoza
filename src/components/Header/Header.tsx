const Header = () => {
  return (
    <header id="header">
      <div className="logo">ğŸ¥Ÿ</div>
      <div className="logo-text">gyoza</div>
    </header>
  );
};

export default Header;
