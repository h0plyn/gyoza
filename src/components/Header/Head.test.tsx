import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('<Header />', () => {
  afterEach(cleanup);

  const setup = () => {
    const utils = render(<Header />);
    return { ...utils };
  };

  it('should render the Header component without crashing', () => {
    render(<Header />);
  });

  it('matches the snapshot of Header', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an ðŸ¥Ÿ emoji icon and title', () => {
    const { getByText } = setup();
    expect(getByText('ðŸ¥Ÿ')).toBeInTheDocument();
    expect(getByText('gyoza')).toBeInTheDocument();
  });
});
