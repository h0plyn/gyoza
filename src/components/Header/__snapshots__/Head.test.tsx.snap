// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches the snapshot of Header 1`] = `
<header
  id="header"
>
  <div
    className="logo"
  >
    ðŸ¥Ÿ
  </div>
  <div
    className="logo-text"
  >
    gyoza
  </div>
</header>
`;
